clang/lib/Sema/SemaExpr.cpp

```diff
@@ -226,6 +226,13 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                !IsConst && VD->hasGlobalStorage() && !isa<ParmVarDecl>(VD))
         SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
             << Sema::KernelGlobalVariable;
+      // Using ESIMD globals in SYCL context is not allowed
+      else if (IsRuntimeEvaluated && isSYCLEsimdPrivateGlobal(VD) &&
+               VD->hasGlobalStorage() && !isa<ParmVarDecl>(VD) &&
```

Comment:
Under what circumstances can the variable have global storage as a parameter? e.g., can we drop the `!isa<ParmVarDecl>` because the global storage already implies that?

Reply:
Done.